client/src/unifyfs.c

```diff
@@ -1105,6 +1105,11 @@ int unifyfs_fid_truncate(int fid, off_t length)
     unifyfs_filemeta_t* meta = unifyfs_get_meta_from_fid(fid);
     assert(meta != NULL);
 
+    /* truncate is not valid for directories */
+    if (S_ISDIR(meta->mode)) {
+        return EISDIR;
+    }
+
```

Comment:
We don't want to fix this here, but this code sequence reminds me that we still have some work to do to make sure our local cache of metadata is consistent after unlink.  For example in this case, I think the following sequence will cause problems for us:
```
rank 0:
mkdir("/unifyfs/foo")

rank 1:
remove("/unifyfs/foo")
fd = open("/unifyfs/foo", O_CREAT)
write(fd)
close(fd)

// at this point, the directory foo has been replaced by a file so truncate should work,
// but the cache in rank 0 will still see it as a directory so it will fail
rank 0:
truncate("/unifyfs/foo")
```